refactor(transformer/object-rest-spread): extract unboxed field earlier

overlookmotel opened the pull request:
Pure refactor. Follow-on after #12698.

We only need the `target` field, so no need bring the whole of `AssignmentTargetRest` onto the stack. Possibly compiler will have optimized this already, but better to be explicit and make it easier for compiler to apply the optimization.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12717?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#12717** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12717?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12717?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#12716** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12716?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12717).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing reviewed (approved):
I've noticed this before, but Boshen merged that PR so quickly.

overlookmotel committed:
refactor(transformer/object-rest-spread): extract unboxed field earlier (#12717)

Pure refactor. Follow-on after #12698.

We only need the `target` field, so no need bring the whole of `AssignmentTargetRest` onto the stack. Possibly compiler will have optimized this already, but better to be explicit and make it easier for compiler to apply the optimization.
